### DISCUSSION TITLE
Fritzbox 7530 no "add swap" option.

### DISCUSSION THREAD

**Discussion by ektorasdj** [Freetz-NG/freetz-ng#233] [Allgemein] [answered]
Hello and thank you for this fine tool you created.
I have one problem that i dont think its intended:
When i select the 7530 device in the menuconfig the add swap option disappears..
Freetz-ng Latest trunk version 18155 for 7.25 fritzos.

**Comment by fda77** [comment]
AVM has disbled CONFIG_SWAP support for ipq40xx & puma7 in the kernel. So it wont work!
See the kernel-config files in make/linux/configs/avm/
To enable it, a self build kernel (replace kernel) is required - but this is not available yet with freetz

**Reply by fda77** [reply]
See also https://github.com/Freetz/freetz/issues/165

**Comment by ektorasdj** [comment]
I see.... Thank you for answering so quickly . Too bad.... Unfortunately i am not so proficient to create a self build kernel. I hope avm re-enables support on future kernels.
Thank you again, and everything that freetz has to offer is awesome!

**Comment by fda77** [comment, marked as answer]
I dotn think AVM will enable it as ist for this devices is always disabled. So somebody need to do replace-kernel support

**Reply by ektorasdj** [reply]
Hmmm.... My 7530 reports only 14mb free ram , 120 used(+85 cache). Do you think a swap file on usb would have helped if it was supported? I hope someone will get on it some day.

**Comment by fda77** [comment]
Sure. Swap on USB or internl storage is slower than real memory, but if it is otherwise not enough ram
